packages/cli-kit/src/private/node/api/graphql.ts

```diff
@@ -23,10 +23,22 @@ export function graphqlRequest<T>(
     const clientOptions = {agent: await httpsAgent(), headers}
     const client = new GraphQLClient(url, clientOptions)
     const t0 = performance.now()
-    const response = await client.request<T>(query, variables)
+    const response = await client.rawRequest<T>(query as string, variables)
     const t1 = performance.now()
-    outputDebug(`Request to ${url.toString()} completed in ${Math.round(t1 - t0)} ms`)
-    return response
+
+    const responseHeaders: {[key: string]: string} = {}
+    const interestingHeaders = new Set(['cache-control', 'content-type', 'etag', 'x-request-id'])
+    response.headers.forEach((value, key) => {
+      if (interestingHeaders.has(key)) responseHeaders[key] = value
+    })
```

Comment:
There's [another function](https://github.com/Shopify/cli/blob/8b7c9494071783c9da4b67c726109460c218ca8d/packages/cli-kit/src/public/node/http.ts#L76) to interact with Shopify APIs. What about making the logic in this function reusable and share it across the two functions?